autopilot/autopilot.go

```diff
@@ -31,9 +31,8 @@ type Bus interface {
 	WalletSign(txn *types.Transaction, toSign []types.OutputID, cf types.CoveredFields) error
 
 	// hostdb
-	AllHosts() ([]hostdb.Host, error)
-	Hosts(notSince time.Time, max int) ([]hostdb.Host, error)
 	Host(hostKey consensus.PublicKey) (hostdb.Host, error)
+	Hosts(offset, limit int, whitelist []string) ([]hostdb.Host, error)
```

Comment:
A whitelist for hosts should be a list of pubkeys. That way a host that migrates between IPs or domains will still be on it as long as the pub key doesn't change. It also prevents edge cases like hosts announcing themselves under URLs they don't own which are known to be on a whitelist. 

The autopilot could expose an endpoint which allows for adding a URL to the whitelist which is then resolved to a pubkey right away and stored.

Overall I'm skeptical of a whitelist param since the whole `IN (?)` SQL condition doesn't scale as we know. Since the query gets too long. So maybe the better approach would be a column in the hostdb which is either set to 'true' or 'false' and then implicitly filtered for.